cmd/hlb/command/module.go

```diff
@@ -134,16 +143,28 @@ func Vendor(ctx context.Context, cln *client.Client, info VendorInfo) (err error
 		// Handle diagnostic errors.
 		spans := diagnostic.Spans(err)
 		for _, span := range spans {
-			fmt.Fprintf(info.ErrOutput, "%s\n", span.Pretty(ctx))
+			fmt.Fprintf(info.Stderr, "%s\n", span.Pretty(ctx))
 		}
 
 		err = errdefs.WithAbort(err, len(spans))
 	}()
 
-	ctx = filebuffer.WithBuffers(ctx, builtin.Buffers())
-	mod, err := parser.Parse(ctx, rc)
+	mod, err := ParseModuleURI(ctx, cln, info.Stdin, uri)
 	if err != nil {
-		return err
+		if !os.IsNotExist(err) {
+			return err
+		}
+
+		rc, err := findVendoredModule(err, uri)
+		if err != nil {
+			return err
+		}
+		defer rc.Close()
```

Comment:
Is it intentional to delay the close until the function returns, instead of only after `parser.Parse`?